torch/jit/frontend.py

```diff
@@ -305,6 +305,32 @@ def get_default_args(fn):
     }
 
 
+def get_default_args_for_class(cls):
+    """
+    Get default arguments for all methods in a class (except for static methods).
+
+    Args:
+        cls: type - The class type to inspect for default arguments.
+    Returns:
+        A Dict[str, Dict[str, Any]] which maps each method name to a Dict[str, Any]
+        that maps each argument name to its default value.
+    """
+    # Get methods (except static methods because those are compiled separately as
+    # if they were independent script functions).
+    methods = inspect.getmembers(
+        cls,
+        predicate=lambda m: (inspect.ismethod(m) or inspect.isfunction(m))
+        and not is_static_fn(cls, m.__name__)
+        and m.__name__ in cls.__dict__
+    )
+
+    # Get method defaults. Property defaults do not need to be considered
+    # because setters cannot be invoked without a value.
+    defaults = {method[0]: get_default_args(method[1]) for method in methods}
```

Comment:
nit maybe: `defaults = {method_name: get_default_args(method_impl)) for method_name, method_impl in methods}` to be a little more readable

Reply:
Woah, I didn't know you unpack tuples like that inside a comprehension but it makes sense. Cool!